### DISCUSSION TITLE
Reusable form component

### DISCUSSION THREAD

**Discussion by TheoThmn** [remsfal/remsfal-frontend#285] [Ideas]
To create consistency across the create/update views, I believe it's important that we agree on a standardized format for displaying these views. With this in mind, I propose using modals (specifically the [PrimeVue Modal](https://primevue.org/dialog/#modal)) that overlay the application when creating or editing an asset. This approach would ensure a consistent user experience and allow users to focus on the task at hand without distractions.

Specifically, I think the content of the modal could follow a similar design to [src/components/NewProjectForm.vue](https://github.com/remsfal/remsfal-frontend/blob/main/src/components/NewProjectForm.vue), as it seems to be the established structure for form-based interactions in this project.

This discussion relates to issues #230, #231, #232, #233, and #234, so I would love to hear your thoughts on this, @astanik! I have not seen modals in the rest of the application. Would this approach align with the overall goals for these features, or do you have any suggestions for improvements?

**Comment by ArniBananii** [comment]
I agree with this statement. Modals dont remove you as much from the flow as much as a full page view might do.

**Comment by TheoThmn** [comment]
Would it be sensible to make a default modal view on which all of the aforementioned issue-assignees can rely on? This could enhance consistency over all of the modals that might follow in remsfal.

**Reply by ZieglerAaron** [reply]
It would probably make the most sense if we had some kind of standard modal, if we want to implement that now for our respective KBE's we should agree on a form, for me it would be totally fine to stick to your suggestion in the link.

**Comment by TheoThmn** [comment]
After discussing with @basselali97, @ZieglerAaron, and @ChristopherDKeller (#218), we have decided on the following approach for the implementation:

1. The implementation for #218 will utilize a standard view components and pass the Asset type and ID to the view. The component will handle showing the right view.
2. For new asset creations, we will use modals, as suggested in #295.
3. We will add an full page for updating assets, ensuring that the "Submit" button is only enabled when values have been changed.
4. We will not implement a separate view for viewing asset information. Users will have to use the update view to check asset details.

I will work on the standard view component for the creation of new assets and the standard view component for updating assets.
Let me know if anyone has any further thoughts!

**Reply by mesyamaureen** [reply]
@TheoThmn @basselali97 @ZieglerAaron @ChristopherDKeller have you guys decided, who's going to implement the reusable modal component? or how detailled have you guys disccused this issue?
It would be great, if you could invite me to your meetings because my issue #233 touches this issue as well :)

**Reply by TheoThmn** [reply]
Hello, thank you for your message. @ZieglerAaron and I will make a proposal for the reusable modal component (and optionally the reusable form component) as soon as possible. We know that all of you rely on it.
As for the meetings I've contacted you via LinkedIn.

**Reply by TheoThmn** [reply]
Hello everybody,

I have created the reusable form component. Please head to the [corresponding branch on my fork](https://github.com/TheoThmn/remsfal-frontend/blob/reusable-form-component/src/components/ReusableFormComponent.vue). I have implemented the following:

- input a custom headline as property.
- input custom text on the submit and cancel buttons as property.
- select from the following field options: 
  - text
  - textarea
  - checkbox
  - dropdown
- furthermore you may give the following inputs for the form-fields
  - name
  - label
  - options for dropdowns
  - required (boolean)
- may pass on initial values (for the update views)
- pass on an array of validation functions.
- pass on how to handle clicks on the submit- and cancel-button.

<img width="624" alt="image" src="https://github.com/user-attachments/assets/7d95440e-d84d-4fb3-98a4-c3db5f151ab8">


Also note that we have agreed with @astanik, that we would route in this format
` /project/:projectId/commercial/:commercialId`
instead of this format
` /project/:projectId/building/:buildingId/commercial/:commercialId`

**Reply by TheoThmn** [reply]
Hello everyone,
after testing I've found some minor issues with the first draft of the component. These were addressed in the newest pull request #321 that is now awaiting review. Please update your code, if you've already used the component in your branches.

**Comment by astanik** [comment]
@Hannesson1 This discussion may be interresting for you. #235

**Reply by Hannesson1** [reply]
Thank you, I will check and update the routes.

**Comment by ChristopherDKeller** [comment]
I wanted to clarify the handling of creating and editing entities (e.g. property, building, etc.).  When an entity is edited, the following path is called: `/project/${projectId}/${entity (e.g.  property, building, etc.)}/${entityId}`.
When an entity is created, the following path is called: `/project/${projectId}/${entity}/create?${parentId}`,
with the parent ID as a query parameter. E.g. if entity = building, the query parameter is the ID of the parent property.

**Comment by astanik** [comment]
I have created a new issue for handling with short URLs at the backend: https://github.com/remsfal/remsfal-backend/issues/275

Thus, you can make your request on the same way as in the frontend and will be redirected to the long nested subresource.
